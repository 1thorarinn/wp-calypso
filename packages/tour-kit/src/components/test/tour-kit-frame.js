/**
 * @jest-environment jsdom
 */
import { shallow } from 'enzyme';
import KeyboardNavigation from '../keyboard-navigation';
import TourKitFrame from '../tour-kit-frame';

const defaultConfig = {
	steps: [
		{
			referenceElements: null,
		},
	],
	renderers: {
		tourStep: jest.fn( () => <div className="test__tour" /> ),
		tourMinimized: jest.fn( () => <div className="test__minimized" /> ),
	},
	closeHandler: jest.fn( () => {} ),
	options: {
		className: 'optional-classname',
		effects: {
			overlay: false,
		},
	},
};

describe( 'TourKitFrame', () => {
	test( 'should render', () => {
		const wrapper = shallow( <TourKitFrame config={ defaultConfig } /> );
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should have right classNames', () => {
		const wrapper = shallow( <TourKitFrame config={ defaultConfig } /> );
		expect( wrapper.find( '.tour-kit-frame' ) ).toHaveLength( 1 );
		expect( wrapper.find( '.tour-kit-frame' ).hasClass( 'optional-classname' ) ).toEqual( true );
	} );

	test( 'should have keyboard navigation', () => {
		const wrapper = shallow( <TourKitFrame config={ defaultConfig } /> );
		expect( wrapper.find( KeyboardNavigation ) ).toHaveLength( 1 );
	} );

	test( 'should render right renderer', () => {
		const wrapper = shallow( <TourKitFrame config={ defaultConfig } /> );
		const stepRenderer = wrapper.find( '.test__tour' );
		expect( stepRenderer ).toHaveLength( 1 );
		expect( defaultConfig.renderers.tourStep ).toHaveBeenCalled();
		expect( wrapper.find( '.test__minimized' ) ).toHaveLength( 0 );

		const minimizeFunction = wrapper.find( KeyboardNavigation ).prop( 'onMinimize' );
		minimizeFunction();

		const minimizedRenderer = wrapper.find( '.test__minimized' );
		expect( minimizedRenderer ).toHaveLength( 1 );
		expect( wrapper.find( '.test__tour' ) ).toHaveLength( 0 );
	} );
} );
