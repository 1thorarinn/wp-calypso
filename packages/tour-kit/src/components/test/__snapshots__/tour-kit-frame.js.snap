// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourKitFrame should render 1`] = `
<Fragment>
  <KeyboardNavigation
    isMinimized={false}
    onDismiss={[Function]}
    onMinimize={[Function]}
    onNextStepProgression={[Function]}
    onPreviousStepProgression={[Function]}
    tourContainerRef={
      Object {
        "current": null,
      }
    }
  />
  <div
    className="tour-kit-frame optional-classname"
  >
    <div
      className="tour-kit-frame__container"
    >
      <div
        className="test__tour"
      />
    </div>
  </div>
</Fragment>
`;
